### PR TITLE
add full pre hook that calls k8s bootstrap and setting k8s credentials

### PR DESCRIPTION
**Pull request by johrstrom**
Add full pre hook that calls k8s bootstrap and setting k8s credentials.

### CODE REVIEW

**Comment by johrstrom** [comment]
Most of the updates in `hooks/k8s-bootstrap.sh` are to rename USERNAME to ONDEMAND_USERNAME. But also quoting strings just after seeing shellcheck output (which we should start using in our rake tests).

File: hooks/set-k8s-creds.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# This file has environment specific stuff like TOKEN_URL, CLIENT_SECRET,
+# and K8S_USERNAME
+# shellcheck disable=SC1091
+source "/opt/osc/etc/ondemand-k8.env"
+
+# we use pass ACCESS_TOKEN into the id-token arg. That's OK, it works and refreshes.
+sudo -u "$ONDEMAND_USERNAME" kubectl config set-credentials "$K8S_USERNAME" \
+   --auth-provider=oidc \
+   --auth-provider-arg=idp-issuer-url="$TOKEN_URL" \
+   --auth-provider-arg=client-id="ondemand" \
```
Comment (treydock) [inline hooks/set-k8s-creds.sh:18]: That client ID is not going to work.  The client ID for each OnDemand instance is the same as the `servername` in OOD configs so for dev it's `ondemand-dev.osc.edu`. I think that is one thing we will have to put into `/opt/osc/etc/ondemand-k8.env`

Comment (treydock) [reply hooks/set-k8s-creds.sh:18]:
```suggestion
   --auth-provider-arg=client-id="$CLIENT_ID" \
```

Comment (johrstrom) [reply hooks/set-k8s-creds.sh]: Yep. added.

**Pull request merged**